### PR TITLE
Add sdlpal.

### PR DESCRIPTION
**Pull request by feserr**
# New Port for Legend of sword and fairy

## Game Information
- **Title**: The Legend of Sword and Fairy
- **URL**: https://github.com/sdlpal/sdlpal

Thanks to Cebion, tabreturn and Ganimoth for helping testing and reviewing it at Discord.

### CFW Tests
Ensure your game has been tested on all major CFWs:
- [x] AmberELEC
- [x] ArkOS
- [x] ROCKNIX
- [x] muOS

### Architecture
- [x] ARM 64
- [x] ARM 32
- [x] x86_64

### Resolution Tests
Test all major resolutions:
- [x] 480x320
- [x] 640x480 (Renders at 640x400)
- [x] 720x720 (Renders at 640x400)
- [x] Higher resolutions (Renders at 640x400)

### CODE REVIEW

File: ports/sdlpal/The Legend of Sword and Fairy.sh [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+else
+  controlfolder="/roms/ports/PortMaster"
+fi
+
+source $controlfolder/control.txt
+source $controlfolder/device_info.txt
+
+[ -f "${controlfolder}/mod_${CFW_NAME}.txt" ] && source "${controlfolder}/mod_${CFW_NAME}.txt"
+
+if [ -f "${controlfolder}/libgl_${CFW_NAME}.txt" ]; then
```
Comment (kloptops) [inline ports/sdlpal/The Legend of Sword and Fairy.sh:20]: This needs to be done after you cd into $GAMEDIR:
```suggestion
get_controls

GAMEDIR=/$directory/ports/sdlpal
cd $GAMEDIR

if [ -f "${controlfolder}/libgl_${CFW_NAME}.txt" ]; then
```

File: ports/sdlpal/The Legend of Sword and Fairy.sh [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  source "${controlfolder}/libgl_${CFW_NAME}.txt"
+else
+  source "${controlfolder}/libgl_default.txt"
+fi
+
+get_controls
+
+GAMEDIR=/$directory/ports/sdlpal
+cd $GAMEDIR
```
Comment (kloptops) [inline ports/sdlpal/The Legend of Sword and Fairy.sh:29]: Remove the `$GAMEDIR` declaration and `cd` since they will now be done above.

**Review by kloptops — CHANGES_REQUESTED** [review]

File: ports/sdlpal/The Legend of Sword and Fairy.sh (hunk 1 above)
Comment (feserr) [reply ports/sdlpal/The Legend of Sword and Fairy.sh:25]: Done.

File: ports/sdlpal/The Legend of Sword and Fairy.sh (hunk 2 above)
Comment (feserr) [reply ports/sdlpal/The Legend of Sword and Fairy.sh]: Done.

**Review by kloptops — APPROVED** [review]
👍

**Review by Cebion — APPROVED** [review]

**Pull request merged**